src/foam/core/OrX.java

```diff
@@ -0,0 +1,56 @@
+/**
+ * @license
+ * Copyright 2021 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package foam.core;
+
+/** Simple Or X implementation.
+    get - first test delegate, then local
+    put - put to delegate
+ **/
+public class OrX
```

Comment:
I think the put() semantics is incorrect. If you put(), it will go to the delegate, which means you can't override any values from localX_. I think on put() you would need to do something like:
localX_ = localX_.put().

Reply:
The OR'd contexts are like a linked list. 
root -> delegate -> delegate
get - lookup in delegate first, working it's way back to root
put - puts into delegate

Reply:
ah, my initial version and documentation was reversed.